src/main/java/mat/server/service/FhirValidationReportService.java

```diff
@@ -95,31 +102,78 @@ private String generateValidationReport(ConversionResultDto conversionResultDto,
+                if (CollectionUtils.isNotEmpty(conversionResultDto.getValueSetConversionResults())) {
+                    for (ValueSetConversionResults valueSetConversionResults : conversionResultDto.getValueSetConversionResults()) {
+                        if (Boolean.FALSE.equals(valueSetConversionResults.getSuccess())) {
+                            valueSetFhirValidationErrors = valueSetConversionResults.getValueSetFhirValidationResults();
+                        }
+                    }
+                }
+
+                // Library FHIR validation errors
+                List<FhirValidationResult> libraryFhirValidationErrors = new ArrayList<>();
+                Set<CqlConversionError> qdmCqlConversionErrors = new HashSet<>();
+                Set<MatCqlConversionException> qdmMatCqlConversionErrors = null;
+                Set<CqlConversionError> fhirCqlConversionErrors = null;
+                Set<MatCqlConversionException> fhirMatCqlConversionErrors = null;
+                if (CollectionUtils.isNotEmpty(conversionResultDto.getLibraryConversionResults())) {
+                    for (LibraryConversionResults results : conversionResultDto.getLibraryConversionResults()) {
+                        if(results.getReason().equalsIgnoreCase("Not Found in Hapi")) {
+                            paramsMap.put("LibraryNotFoundInHapi", results.getReason());
+                        }
+                        if (Boolean.FALSE.equals(results.getSuccess())) {
+                            if (CollectionUtils.isNotEmpty(results.getLibraryFhirValidationResults())) {
+                                libraryFhirValidationErrors = results.getLibraryFhirValidationResults();
+                            }
+                            // CQL conversion errors
+                            if (results.getCqlConversionResult() != null) {
+                                if (CollectionUtils.isNotEmpty(results.getCqlConversionResult().getCqlConversionErrors())) {
+                                    qdmCqlConversionErrors = results.getCqlConversionResult().getCqlConversionErrors();
+                                }
+                                if (CollectionUtils.isNotEmpty(results.getCqlConversionResult().getMatCqlConversionErrors())) {
+                                    qdmMatCqlConversionErrors = results.getCqlConversionResult().getMatCqlConversionErrors();
+                                }
+                                if (CollectionUtils.isNotEmpty(results.getCqlConversionResult().getFhirCqlConversionErrors())) {
+                                    fhirCqlConversionErrors = results.getCqlConversionResult().getFhirCqlConversionErrors();
+                                }
+                                if (CollectionUtils.isNotEmpty(results.getCqlConversionResult().getFhirMatCqlConversionErrors())) {
+                                    fhirMatCqlConversionErrors = results.getCqlConversionResult().getFhirMatCqlConversionErrors();
```

Comment:
Can we reduce these repeated calls?
e.g. `results.getCqlConversionResult()` is called 7-8 times in this method. we can assign the o/p from this call to some variable and re-use it.

Reply:
Will make the change

Reply:
Done